src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -228,9 +229,10 @@ private UserDetails retrieveUser(String username, UsernamePasswordAuthentication
     @SuppressFBWarnings(value = "ES_COMPARING_PARAMETER_STRING_WITH_EQ", justification = "Intentional instance check.")
     public UserDetails retrieveUser(final String username, final String password, final String domainName, final List<SocketInfo> ldapServers) {
         UserDetails userDetails;
+        String hashKey = DigestUtils.sha1Hex(username + "@@" + password);
```

Comment:
there is a possibility of collisions here - even if very small.  it would be better to leave the username component as-is this will be unique, and then to hash the password so that the key will be unique.

`String hashKey = username + "@@" + DigestUtils.sha1Hex(password);`